crates/evm/src/tests/sys_tx_tests.rs

```diff
@@ -0,0 +1,165 @@
+        nonce: 0,
+        storage: [
+            (U256::from_be_slice(&hex!("0000000000000000000000000000000000000000000000000000000000000000")), U256::from_be_slice(SYSTEM_SIGNER.into_word().as_slice())),
+        ].into_iter().collect(),
+    });
+
+    let (evm, mut working_set) = get_evm(&config);
+
+    assert_eq!(
+        evm.receipts
+            .iter(&mut working_set.accessory_state())
+            .collect::<Vec<_>>(),
+        [
+            Receipt { // L1BlockHashList::initializeBlockNumber(U256)
+                receipt: reth_primitives::Receipt {
+                    tx_type: reth_primitives::TxType::Eip1559,
+                    success: true,
+                    cumulative_gas_used: 45711,
+                    logs: vec![]
+                },
+                gas_used: 45711,
+                log_index_start: 0,
+                diff_size: 284,
+                error: None
+            },
+            Receipt { // L1BlockHashList::setBlockInfo(U256, U256)
+                receipt: reth_primitives::Receipt {
+                    tx_type: reth_primitives::TxType::Eip1559,
+                    success: true,
+                    cumulative_gas_used: 121376,
+                    logs: vec![
+                        Log {
+                            address: L1BlockHashList::address(),
+                            topics: vec![b256!("32eff959e2e8d1609edc4b39ccf75900aa6c1da5719f8432752963fdf008234f")],
+                            data: Bytes::from_static(&hex!("000000000000000000000000000000000000000000000000000000000000000201010101010101010101010101010101010101010101010101010101010101010202020202020202020202020202020202020202020202020202020202020202")),
+                        }
+                    ]
+                },
+                gas_used: 75665,
+                log_index_start: 0,
+                diff_size: 412,
+                error: None
+            },
+        ]
+    );
+
+    let l1_fee_rate = 1;
+
+    let deploy_message =
+        create_contract_message_with_fee(&dev_signer, 0, BlockHashContract::default(), 1);
```

Comment:
either should be below where this is used or just call the function inside the vec